scalafix/rules/src/main/scala/rsc/rules/pretty/SemanticdbPrinter.scala

```diff
@@ -81,6 +82,43 @@ class SemanticdbPrinter(env: Env, index: DocumentIndex, config: RscCompatConfig)
     case _ => sys.error(s"unsupported tree $tree")
   }
 
+  def symbolOf(signature: Signature): Option[String] =
+    signature match {
+      case TypeSignature(_, lower: TypeRef, upper: TypeRef) if lower == upper => Some(lower.symbol)
+      case _ => None
+    }
+
+
+  def symIsInScope(sym: String, env: Env): Boolean = {
+    extractName(sym)
+      .map(env.lookup) match {
+      // TODO: If the lookup returns NoSymbol, we can insert an import and skip the prefix.
+      // The logic to implement this is left for future work.
+      case None => false
+      case Some(symFromScope) => isSameOrAlias(sym, symFromScope)
+    }
+  }
+
+  def isSameOrAlias(sym: String, envSym: String) =
+    (sym == envSym) || dealias(envSym).contains(sym)
+
+  def dealias(sym: String): Option[String] = {
+    index.symbols.get(sym).flatMap(s => symbolOf(s.signature))
+  }
+
+  def extractName(sym: String): Option[Scala.Names.Name] = {
```

Comment:
The idiomatic way of referring to `Name` is `import scala.meta.internal.semanticdb.Scala.{Names => n}` and then `n.Name`.